seedsource_core/django/seedsource/management/commands/calculate_zone_transfers.py

```diff
@@ -7,268 +7,369 @@
+    get_region_for_zone,
+    calculate_pixel_area,
 )
+from seedsource_core.django.seedsource.management.dataset import (
+    ElevationDataset,
+    ClimateDatasets,
+)
+from seedsource_core.django.seedsource.management.zoneconfig import ZoneConfig
+
+
+PERIODS = ("1961_1990", "1981_2010")
 
 
 class Command(BaseCommand):
-    help = 'Calculates default variable transfer limits for each available seed zone'
+    help = "Calculates default variable transfer limits for each available seed zone"
 
     def __init__(self, *args, **kwargs):
         self.transfers_by_source = {}
 
         super().__init__(*args, **kwargs)
 
     def add_arguments(self, parser):
-        parser.add_argument('source_name', nargs='?', type=str, default='all', help='Accepts zone name or all')
-
-    def _get_subsets(self, elevation, data, coords: SpatialCoordinateVariables, bbox):
-        """ Returns subsets of elevation, data, and coords, clipped to the given bounds """
 
-        x_slice = slice(*coords.x.indices_for_range(bbox.xmin, bbox.xmax))
-        y_slice = slice(*coords.y.indices_for_range(bbox.ymin, bbox.ymax))
+        parser.add_argument(
+            "--zones",
+            dest="zoneset",
+            default=None,
+            help="Comma delimited list of zones sets to analyze. (default is to analyze all available zone sets)",
+        )
 
-        if x_slice.stop - x_slice.start < 2:
-            x_slice = slice(x_slice.start, x_slice.start + 2)
-        if y_slice.stop - y_slice.start < 2:
-            y_slice = slice(y_slice.start, y_slice.start + 2)
+        parser.add_argument(
+            "--variables",
+            dest="variables",
+            default=None,
+            help="Comma delimited list of variables analyze. (default is to analyze all available variables)",
+        )
 
-        return elevation[y_slice, x_slice], data[y_slice, x_slice], coords.slice_by_window(Window(y_slice, x_slice))
+        parser.add_argument(
+            "--clear",
+            dest="clear",
+            action="store_true",
+            default=False,
+            help="If True, will automatically delete all transfer limits before calculating new ones",
+        )
 
-    def _write_limit(self, variable, time_period, zone, masked_data, low=None, high=None, label=None):
-        min_value = numpy.nanmin(masked_data)
-        max_value = numpy.nanmax(masked_data)
+    def _write_limit(
+        self, variable, time_period, zone, masked_data, band, elevation_service
+    ):
+        min_value = masked_data.min()
+        max_value = masked_data.max()
         transfer = (max_value - min_value) / 2.0
         center = max_value - transfer
 
-        if numpy.isnan(transfer) or hasattr(transfer, 'mask'):
-            self.stdout.write('WARNING: Transfer limit is NaN for {}, zone {}, band {}-{}'.format(
-                zone.zone_source.name, zone.zone_id, low, high
-            ))
-            return
```

Comment:
This check is important. There are some zone sets for which this case is true, and this prints warnings rather than bailing on the whole command.

Reply:
Because the data are extracted out only within valid areas prior to calling this function, I didn't see a case where this would occur as now written.

Everything worked OK for all zones except the provisional seed zones, for which I don't have enough data to test.

Do you recall a zone set that produced this result?

Reply:
This check wasn't here for invalid areas, it was for elevation bands which have no pixels. E.g., a zone covers pixels with elevation from 1000 - 2000ft, but one rule is for elevations 0 - 500 ft.

Reply:
That could be captured now by [this check](https://github.com/consbio/seedsource-core/pull/31/files#diff-c7f00b3027199702d660563fefa8e09aR323) prior to calling `_write_limit()`.  It just isn't emitting a warning, since this is not a user-facing warning: there just isn't data within a particular zone / elevation band combination.